### ISSUE TITLE
Go/JavaScript: codegen probably didn't mean to limit ATN segment size to 2 XOR 31

### ISSUE THREAD

**Issue by dgryski** [antlr/antlr4#2629]
The fix from https://github.com/antlr/antlr4/pull/1805 needs to be applied to both the JavaScript and Go targets.

```
github.com/antlr/antlr4/tool/src/org/antlr/v4/codegen/target/JavaScriptTarget.java:91:		return 2 ^ 31;
github.com/antlr/antlr4/tool/src/org/antlr/v4/codegen/target/GoTarget.java:137:		return 2 ^ 31;
```

**Comment by KvanTTT** [comment]
Would you like to suggest a fix (PR) for that?

**Comment by dgryski** [comment]
This was fixed in 3a5f112c3295c1aeb9414919d0662232f4a97f6e and 24d9128d96b6dad1b7589ac63f698e1aac35720a
